src/vnode.ts

```diff
@@ -12,22 +14,32 @@ export interface VNode {
 }
 
 export interface VNodeData {
-  // modules - use any because Object type is useless
-  props?: any;
-  attrs?: any;
-  class?: any;
-  style?: any;
-  dataset?: any;
-  on?: any;
-  hero?: any;
-  attachData?: any;
+  props?: {
+    [prop: string]: any
+  };
+  attrs?: {
+    [attr: string]: any
+  };
+  class?: {
+    [name: string]: boolean
+  };
+  style?: VNodeStyle;
+  dataset?: {
+    [name: string]: string
+  };
+  on?: {
+    [event: string]: EventListener | EventListener[]
```

Comment:
This type is not correct. When given an array only the first element has to be an event listener. Changing `EventListener[]` to `any[]` would probably be the easiest fix. I at least can't think of a better way right now.